includes/wf_crm_webform_base.inc

```diff
@@ -186,6 +190,36 @@ abstract class wf_crm_webform_base {
   }
 
   /**
+   * Reorder returned results according to settings chosen in wf_civicrm backend
+   *
+   * @param string $ent
+   * @param array $values
+   * @return array $reorderedArray
+   * @todo Handle user-select case
+   */
+  protected function reorderByLocationType($ent, $values = array()){
+    $reorderedArray = array();
+    // For every location field setting
+    foreach ($this->settings['data']['contact'] as $setting) {
+      // For every Entity
+      foreach($setting[$ent] as $e){
+        // For every result row
+        foreach($values as $key => $value){
+          if((in_array($ent, array('address', 'email')) && $value['location_type_id'] == $e['location_type_id'])
+            ||
+            ($value['location_type_id'] == $e['location_type_id'] && $value[$ent.'_type_id'] == $e[$ent.'_type_id'])     
+            ){
+              $reorderedArray[$key] = $value;
+              unset($values[$key]);
+              break;
```

Comment:
it appears the effect of this will be to take an array with potentially multiple values and return an array with at most one value. This would cause the contact's second or third phone or address to be "lost"